credential/schema/vcjsonschema_test.go

```diff
@@ -2,58 +2,87 @@ package schema
-
-	vc "github.com/TBD54566975/ssi-sdk/credential"
 )
 
 const (
-	vcJSONTestVector1           string = "vc-json-schema-example-1.json"
-	vcJSONCredentialTestVector1 string = "vc-with-schema-example-11.json"
+	jsonSchema2023Credential1       string = "jsonschema2023-credential-1.json"
+	jsonSchema2023Schema1           string = "jsonschema2023-schema-1.json"
+	credentialSchema2023Credential1 string = "credentialschema2023-credential-1.json"
+	credentialSchema2023Schema1     string = "credentialschema2023-schema-1.json"
 )
 
 var (
 	//go:embed testdata
-	testVectors       embed.FS
-	vcJSONTestVectors = []string{vcJSONTestVector1}
+	testVectors embed.FS
 )
 
-func TestIsValidCredentialSchema(t *testing.T) {
-	for _, tv := range vcJSONTestVectors {
-		schema, err := getTestVector(tv)
+func TestValidateCredentialAgainstSchema(t *testing.T) {
+	t.Run("test validate credential against JsonSchema2023", func(t *testing.T) {
+		cred, err := getTestVector(jsonSchema2023Credential1)
 		assert.NoError(t, err)
-		assert.NoError(t, IsValidCredentialSchema(schema))
-	}
+
+		var vc credential.VerifiableCredential
+		err = json.Unmarshal([]byte(cred), &vc)
+		assert.NoError(t, err)
+
+		err = ValidateCredentialAgainstSchema(&localAccess{}, vc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("test validate credential against CredentialSchema2023", func(t *testing.T) {
+		cred, err := getTestVector(credentialSchema2023Credential1)
+		assert.NoError(t, err)
+
+		var vc credential.VerifiableCredential
+		err = json.Unmarshal([]byte(cred), &vc)
+		assert.NoError(t, err)
+
+		err = ValidateCredentialAgainstSchema(&localAccess{}, vc)
+		assert.NoError(t, err)
+	})
 }
 
-func TestIsCredentialValidForSchema(t *testing.T) {
-	// Load VC
-	credential, err := getTestVector(vcJSONCredentialTestVector1)
-	assert.NoError(t, err)
-	var cred vc.VerifiableCredential
-	err = json.Unmarshal([]byte(credential), &cred)
-	assert.NoError(t, err)
-
-	// Load vcJSONSchema
-	vcJSONSchemaString, err := getTestVector(vcJSONTestVector1)
-	assert.NoError(t, err)
-
-	vcJSONSchema, err := StringToVCJSONCredentialSchema(vcJSONSchemaString)
-	assert.NoError(t, err)
-	assert.NotEmpty(t, vcJSONSchema)
-
-	// Validate credential against vcJSONSchema
-	err = IsCredentialValidForVCJSONSchema(cred, *vcJSONSchema)
-	assert.NoError(t, err)
-
-	// make sure the cred was not modified
-	var credCopy vc.VerifiableCredential
-	err = json.Unmarshal([]byte(credential), &credCopy)
-	assert.NoError(t, err)
-	assert.Equal(t, credCopy, cred)
+type localAccess struct{}
+
+func (localAccess) GetVCJSONSchema(_ VCJSONSchemaType, id string) (JSONSchema, error) {
+	switch id {
+	case "https://example.com/schemas/email.json":
+		schema, err := getTestVector(jsonSchema2023Schema1)
+		if err != nil {
+			return nil, err
+		}
+		var s JSONSchema
+		if err = json.Unmarshal([]byte(schema), &s); err != nil {
+			return nil, err
+		}
+		return s, nil
+	case "https://example.com/schemas/email-credential-schema.json":
```

Comment:
Similar to the comment about the test case setup, this URL should be a credential.

Reply:
switching